packages/plugin-evm/src/providers/wallet.ts

```diff
@@ -175,13 +146,19 @@ export const evmWalletProvider: Provider = {
         message: Memory,
         state?: State
     ): Promise<string | null> {
-        // Check if the user has an EVM wallet
-        if (!runtime.getSetting("EVM_PRIVATE_KEY")) {
+        const privateKey = runtime.getSetting("EVM_PRIVATE_KEY");
+        const chainNames =
+            (runtime.character.settings.chains?.evm as SupportedChain[]) || [];
+
+        if (!privateKey) {
             return null;
         }
 
         try {
-            const walletProvider = new WalletProvider(runtime);
+            const walletProvider = new WalletProvider(
+                privateKey as `0x${string}`,
+                chainNames
+            );
             const address = walletProvider.getAddress();
             const balance = await walletProvider.getWalletBalance();
             return `EVM Wallet Address: ${address}\nBalance: ${balance} ETH`;
```

Comment:
Maybe I'd add the chain you are getting the balance for

Reply:
> Maybe I'd add the chain you are getting the balance for

The current behavior defaults to Ethereum mainnet unless another chain is provided, in which case it uses the first chain in the list (see tests on lines 24-29 and 36-40). Next, we plan to add a switch chain action and a separate action to fetch the balance for a specific chain.

Does this approach make sense?